lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1740,9 +1740,14 @@ ParseResult FIRStmtParser::parseIntegerLiteralExp(Value &result) {
   } else {
     signedness = IntegerType::Unsigned;
     if (width != -1) {
-      // Check for overlow if we are truncating bits.
-      if (unsigned(width) < value.getBitWidth() &&
-          value.countLeadingZeros() < value.getBitWidth() - width) {
+      // Check for overflow if we are truncating bits.
+      if (width == 0) {
+        // The check below already would error on this appropriately,
+        // but check explicitly for symmetry in errors produced.
+        if (!value.isZero())
+          return emitError(loc, "zero bit constant must be zero");
+      } else if (unsigned(width) < value.getBitWidth() &&
+                 value.countLeadingZeros() < value.getBitWidth() - width) {
```

Comment:
Logic looks good! The `if (width == 0)` duplication could be removed by hoisting this out of the `if (signed) / else` grouping.

(Out of scope:) There is a lot of logic that is shared between these outer if/else code paths which could be refactored.